### PR TITLE
added support for binary data in kamus secret

### LINKED ISSUE #246: Unable to add binary file to KamusSecret
In trying to add a DER-encoded certificate to a KamusSecret, it's proven to not be possible. With standard Kubernetes secrets, it would be a case of base64-encoding the file and applying that Secret manifest. With KamusSecret, the encryptor-api expects plaintext (non-base64-encoded) input.

I tried to just pass the DER file raw to the kamus-cli, but I was getting a different output each time it ran. I feel like this should be the correct workflow, but the inconsistent response that it was giving me was not leaving Kubernetes happy.

### PR DESCRIPTION
**Pull request by omerlh**
close #246

### CODE REVIEW

**Comment by omerlh** [comment]
@shaikatz build *finally* passing - can you please take a look so we can start working on merging it in? Need to sharpen some edges before merging...

File: site/content/docs/user/crd.md [hunk 1]
```diff
@@ -27,14 +30,16 @@ kamus-cli encrypt
...
+apiVersion: "soluto.com/v1alpha2"
 kind: KamusSecret
 metadata:
   name: my-tls-secret     //This will be the name of the secret
   namespace: default      //The secret and KamusSecret live in this namespace
 type: TlsSecret           //The type of the secret that will be created
-data:                     //Put here all the encrypted data, that will be stored (decrypted) on the secret data
+stringData:               //Put here all the encrypted data, that will be stored (decrypted) on the secret data
   key: J9NYLzTC/O44DvlCEZ+LfQ==:Cc9O5zQzFOyxwTD5ZHseqg==
+data:                     //Put here base64 encoded data (usually, binary data like private keys in der format)
```
Comment (shaikatz) [inline site/content/docs/user/crd.md]: This is encrypted base 64 right?
I think we need to emphasis that.

File: src/crd-controller/Controllers/ConversionWebhookController.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                                ServiceAccount = sourceKamusSecret.ServiceAccount,
+                                Metadata = sourceKamusSecret.Metadata,
+                                Kind = "KamusSecret",
+                                Type = sourceKamusSecret.Type,
+                                ApiVersion = desiredApiVersion
+                            };
+
+                        default:
+                            Console.WriteLine("Oh no!");
+                            Console.WriteLine(apiVersion);
```
Comment (shaikatz) [inline src/crd-controller/Controllers/ConversionWebhookController.cs]: Can we user the logger and write more descriptive error here?

Comment (omerlh) [reply src/crd-controller/Controllers/ConversionWebhookController.cs]: Yep, fixed - sorry

File: site/content/docs/user/crd.md (hunk 1 above)
Comment (omerlh) [reply site/content/docs/user/crd.md]: fixed

**Comment by omerlh** [comment]
@shaikatz this is finally ready for merging 🎉

File: src/crd-controller/Startup.cs [hunk 3]
```diff
@@ -57,7 +57,8 @@ public Startup(IHostingEnvironment env, IConfiguration config)
             }
             );
                 
-            services.AddHostedService<V1AlphaController>();
+           // services.AddHostedService<V1Alpha1Controller>();
```
Comment (shaikatz) [inline src/crd-controller/Startup.cs]:
```suggestion
```

File: tests/crd-controller/FlowTest.cs [hunk 4]
```diff
@@ -48,11 +49,44 @@ public async Task CreateKamusSecret_SecretCreated()
 
             Assert.Equal("TlsSecret", v1Secret.Type);
             Assert.True(v1Secret.Data.ContainsKey("key"));
-            Assert.Equal("hello", Encoding.UTF8.GetString(v1Secret.Data["key"]));
```
Comment (shaikatz) [inline tests/crd-controller/FlowTest.cs]: Why was that removed?

File: tests/crd-controller/FlowTest.cs [hunk 5]
```diff
@@ -155,7 +191,24 @@ private async Task DeployController()
             Console.WriteLine("Deploying CRD");
             
             RunKubectlCommand("apply -f deployment.yaml");
-            RunKubectlCommand("apply -f crd.yaml");
+
+            Console.WriteLine("running ugly patch");
```
Comment (shaikatz) [inline tests/crd-controller/FlowTest.cs]: Can you add a comment that explains why this patch is required?

File: tests/crd-controller/deployment.yaml [hunk 6]
```diff
@@ -29,6 +37,26 @@ roleRef:
...
   apiGroup: rbac.authorization.k8s.io
 ---
+apiVersion: v1
+kind: Service
+metadata:
+  name: kamus-crd
+  labels:
+    app: kamus
+    component: crd-controller
+    chart: kamus-0.3.0
```
Comment (shaikatz) [inline tests/crd-controller/deployment.yaml]: Can we remove that since we don't install it with chart?
```suggestion
```

File: src/crd-controller/Models/V1Alpha2/KamusSecret.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+{
+    public class KamusSecret : KubernetesObject
+    {
+        public Dictionary<string, string> Data { get; set; }
+        public Dictionary<string, string> StringData { get; set; }
+        public string Type { get; set; }
+        public V1ObjectMeta Metadata { get; set; }
+        public string ServiceAccount { get; set; }
+        
+        public string Status { get; set; }
```
Comment (shaikatz) [inline src/crd-controller/Models/V1Alpha2/KamusSecret.cs]:
```suggestion
```
Looks like it isn't required?

File: src/crd-controller/Models/V1Alpha1/KamusSecret.cs [hunk 8]
```diff
@@ -2,7 +2,7 @@
 using k8s;
 using k8s.Models;
 
-namespace CustomResourceDescriptorController.Models
+namespace CustomResourceDescriptorController.Models.V1Alpha1
```
Comment (shaikatz) [inline src/crd-controller/Models/V1Alpha1/KamusSecret.cs]: Also, remove status if not used

File: tests/crd-controller/FlowTest.cs (hunk 4 above)
Comment (omerlh) [reply tests/crd-controller/FlowTest.cs]: Probably by mistake.

File: tests/crd-controller/FlowTest.cs (hunk 5 above)
Comment (omerlh) [reply tests/crd-controller/FlowTest.cs]: I think it was because `preserveUnknownFields` wasn't supported on k8s version bellow 1.15, but required on 1.15 if a schema is specified. I'll try to remove it and see if we need it anymore.

Comment (omerlh) [reply tests/crd-controller/FlowTest.cs]: I think it was because `preserveUnknownFields` wasn't supported on k8s version bellow 1.15, but required on 1.15 if a schema is specified. I'll try to remove it and see if we need it anymore.

Comment (omerlh) [reply tests/crd-controller/FlowTest.cs]: So... look like this is required - restoring it and adding a comment

**Comment by omerlh** [comment]
@shaikatz let's merge this!
